ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -209,24 +226,31 @@ public static boolean isDisallowedInPullOrScalablePushQueries(
         .anyMatch(col -> col.name.equals(columnName));
   }
 
-  public static int getPseudoColumnVersionFromConfig(final boolean rowpartitionRowoffsetEnabled) {
-    return getPseudoColumnVersionFromConfig(rowpartitionRowoffsetEnabled, false);
+  public static int getPseudoColumnVersionFromConfig(final boolean rowpartitionRowoffsetEnabled,
+      final boolean rowIdEnabled) {
+    return getPseudoColumnVersionFromConfig(rowpartitionRowoffsetEnabled, rowIdEnabled, false);
   }
 
   public static int getPseudoColumnVersionFromConfig(final KsqlConfig ksqlConfig) {
     return getPseudoColumnVersionFromConfig(
         ksqlConfig.getBoolean(KsqlConfig.KSQL_ROWPARTITION_ROWOFFSET_ENABLED),
+        ksqlConfig.getBoolean(KsqlConfig.KSQL_ROWID_ENABLED),
         false
     );
   }
 
   public static int getPseudoColumnVersionFromConfig(
       final boolean rowpartitionRowoffsetEnabled,
+      final boolean rowIdEnabled,
       final boolean forPullOrScalablePushQuery
   ) {
-    return rowpartitionRowoffsetEnabled && !forPullOrScalablePushQuery
-        ? CURRENT_PSEUDOCOLUMN_VERSION_NUMBER
-        : LEGACY_PSEUDOCOLUMN_VERSION_NUMBER;
+    if (rowIdEnabled  && !forPullOrScalablePushQuery) {
```

Comment:
This method is a bit confusing right now because if the ROWID config is enabled, then it doesn't matter whether the ROWPARTITION/ROWOFFSET config is enabled or not (ROWPARTITION and ROWOFFSET are always included if ROWID is enabled). This is fine for now since I plan to clean up (remove) the ROWPARTITION/ROWOFFSET config as part of my toil shift next month.